### ISSUE TITLE
Tx power is limited to only 20 dBm

### ISSUE THREAD

**Issue by kmillerusaf** [cucumber-tony/cucumber-frontend#253]
I'm sure not all APs are the same, but my Ubiquiti UAP AC-Pro is configurable up to 22 dBm when using the UniFi controller. 

I'd like to see this available on Cucumber Tony as well. Doesn't seem like much if you compare 20 to 22 but that's 1.5x the power (100 to 158 mW). Is there a way to intelligently see what each chipset is capable of and allow those thresholds?

Regards,
Keith

**Comment by AndrewMargarit** [comment]
Agree.
Will get added soon.
There still some CRDA related features that we need to work on.

Thanks for reporting!

**Comment by Molkenbur** [comment]
The maximum allowed value for some channels is 27dB (1W).
I would let the user configure up to 27dB on 5 GHz and 20dB for 2.4GHz.
At the end, the device will limit this to the allowed values for the chosen channel.
This will work for auto channel selection, too..

**Comment by AndrewMargarit** [comment]
Yes and no.

When this goes live needs to be properly.
CT needs to be aware of the reg domain that the AP is in and know the rules, displaying only the allowed channels with the associated power output.

The AP will stop broadcasting, without any feedback, if someone adds a radio option that is now allowed in that reg domain.

**Comment by ChristianWinkler** [comment]
@AndrewMargarit : maybe reg domain can be gained from the location data in order to adjust allowed settings ? its already possible to make the AP stop broadcasting by some combinations (channel selecetion for outdoor devices)

**Comment by Molkenbur** [comment]
@AndrewMargarit : I agree. Best is to have the knowledge about the reg domain inside the frontend and guide the user as good as possible.
But to let you know, this is not an easy task. As Christian said, Outdoor APs are limited in the 5 GHz band. A few channels are typically not chosen, because DFS takes 10 minutes (by law). The output power can be up to 1W (27dBm), but only for some channels and only if DFS is active. Dependencies, dependencies, dependencies.

For me it would be still enough to have a generic configuration and let the AP decide if the configuration follows the reg domain and limit the parameters to that. Risk: The AP does not work at all and the operator doesn't know that.
